### PR TITLE
Add a startup probe, if a cluster supports it. 

### PR DESCRIPTION
**Pull request by wzur-r3**
Taken from https://github.com/helm/charts/issues/22279

# What this PR does / why we need it

My Jenkins installation starts for a while, and without setting a high value of master.healthProbeLivenessInitialDelay it restarts a few times.

One way of helping to mitigate this issue is to add a [Startup Probe](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-startup-probes) to the Jenkins deployment. Startup Probes were introduced in K8s v1.16 so it should have a version check

See https://github.com/helm/charts/issues/22279 for an initial comments in previous chart incarnation for another example.

# Checklist

- [x] [DCO](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed

### CODE REVIEW

File: charts/jenkins/templates/jenkins-master-deployment.yaml [hunk 1]
```diff
@@ -240,6 +240,16 @@ spec:
               name: {{ $port.name }}
 {{- end }}
 {{- if .Values.master.healthProbes }}
+{{- if semverCompare ">=1.16-0" .Capabilities.KubeVersion.GitVersion }}
+          startupProbe:
+            httpGet:
+              path: "{{ default "" .Values.master.jenkinsUriPrefix }}/login"
+              port: http
+            initialDelaySeconds: {{ .Values.master.healthProbeStartupInitialDelay }}
```
[the remark this answers is not in the document]
Comment (torstenwalter) [reply charts/jenkins/templates/jenkins-master-deployment.yaml:248]: I would be in favor of `startupProbe` yaml block into values.yaml and render it using tpl function.
This way we could get rid of countless properties to configure a single probe.

I would take it as action item to do the same thing for the other probes for the 3.0.0 release.

**Review by torstenwalter — CHANGES_REQUESTED** [review]

File: charts/jenkins/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: jenkins
 home: https://jenkins.io/
-version: 2.15.3
+version: 2.15.4
```
[the remark this answers is not in the document]
Comment (wzur-r3) [reply charts/jenkins/Chart.yaml:4]: I don't quite follow. It has been changed.

[the remark this answers is not in the document]
Comment (torstenwalter) [reply charts/jenkins/Chart.yaml:4]: I meant changing it to 2.16.0 ;-)

> MAJOR version when you make incompatible API changes,
> MINOR version when you add functionality in a backwards compatible manner, and
> PATCH version when you make backwards compatible bug fixes.

https://semver.org/

[the remark this answers is not in the document]
Comment (wzur-r3) [reply charts/jenkins/Chart.yaml:4]: personally I would leave it to a person who creates an actual release

[the remark this answers is not in the document]
Comment (torstenwalter) [reply charts/jenkins/Chart.yaml:4]: Merging creates a release ;-)

File: charts/jenkins/templates/jenkins-master-deployment.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (wzur-r3) [reply charts/jenkins/templates/jenkins-master-deployment.yaml:248]: I'm not verse enough to replace it with tpl function. I simply used what someone else suggested in the other repo in May

File: charts/jenkins/Chart.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (wzur-r3) [reply charts/jenkins/Chart.yaml]: updated as requested

File: charts/jenkins/templates/jenkins-master-deployment.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (wzur-r3) [reply charts/jenkins/templates/jenkins-master-deployment.yaml]: also `initialDelaySecods` has been removed for startupProbe

**Comment by wzur-r3** [comment]
I'm really sorry for these forced pushes - I keep forgetting about signing my commits...

File: charts/jenkins/templates/jenkins-master-deployment.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (torstenwalter) [reply charts/jenkins/templates/jenkins-master-deployment.yaml]: I created https://github.com/wzur-r3/helm-charts/pull/1 to demonstrate what I meant with using the `tpl` function.

[the remark this answers is not in the document]
Comment (wzur-r3) [reply charts/jenkins/templates/jenkins-master-deployment.yaml]: many thanks for that - I've already merged it your PR into this PR

**Review by torstenwalter — APPROVED** [review]

**Pull request merged**